hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupReaderSchemaHandler.java

```diff
@@ -102,11 +103,24 @@ public FileGroupReaderSchemaHandler(HoodieReaderContext<T> readerContext,
     this.requestedSchema = AvroSchemaCache.intern(requestedSchema);
     this.hoodieTableConfig = metaClient.getTableConfig();
     this.deleteContext = new DeleteContext(properties, tableSchema);
+    this.metaClient = metaClient;
+
+    boolean hasInstantRange = readerContext.getInstantRange().isPresent();
+    boolean shouldAddCompletionTimeField = !metaClient.isMetadataTable()
+        && metaClient.getTableConfig() != null && metaClient.getTableConfig().getTableVersion() != null
+        && metaClient.getTableConfig().getTableVersion().greaterThanOrEquals(HoodieTableVersion.SIX)
```

Comment:
The completion time is only available in version 8 and above if I remember correctly